tools/disasm/disasm.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+
+import argparse
+import collections
+from pathlib import Path
+from typing import BinaryIO
+
+import libyaz0
+import spimdisasm
+from spimdisasm import frontendCommon as fec
+
+from file_addresses import DmaFile, parse_file_addresses, get_z_name_for_overlay
+
+def load_file_splits(
+    context: spimdisasm.common.Context,
+    config_dir:Path,
+    dma_file: DmaFile,
+    f: BinaryIO,
+) -> spimdisasm.mips.FileSplits:
+    f.seek(dma_file.rom_start)
+    if dma_file.rom_end != 0:  # compressed
+        raw_data = f.read(dma_file.rom_end - dma_file.rom_start)
+        data = bytearray(libyaz0.decompress(raw_data))
+    else:
+        data = bytearray(f.read(dma_file.vrom_end - dma_file.vrom_start))
```

Comment:
Decompressing each segment during disassembly doesn't sound like the best idea. Generating a decompressed rom would be better, specially since we'll need it for assets.

Reply:
I would love to get a tool like z64compress up and running now, if we want to. We could just decompress the input rom at the start, right?
 (this purpose aside, itd be helpful for debug-rom modders to have access to compression being included in the build system now)

Reply:
actually idr if z64compress does decompression :D it might not

Reply:
I can setup the compression and decompression, should be easy to take the stuff from MM.
And no, z64compress doesn't do decompression lol

Reply:
I totally agree about compression and decompression but I was hoping to do those in separate PRs. (Although I'd be very happy if Anghelo beats me to it.) Compression at least is not necessary right now, and the decompression code here is 4 lines long so I didn't think it cost that much to add.

Reply:
It isn't much code but it can be impactful on runtime, so separating decompression from disassembly would be a lot better

Reply:
waiting is fine by me too personally, just wanted to express that I am a fan of compression infrastructure in general being introduced

Reply:
I'm not totally sure it would be faster to decompress separately, for the debug ROM at least the uncompressed source files are about 5MB, but decompressing the entire ROM is about 54MB:

```
$ wc -c baserom/{boot,code,ovl_*}
  5091104 total
$ wc -c baserom/*
 54252304 total
```

I actually did try to set up decompression briefly, but I thought it made the build system too complicated, and I wasn't sure if I was heading in the right direction. It might be better to combine that work with multi-version assets

Reply:
(de)compression will be handled in https://github.com/zeldaret/oot/pull/1614
as well as some baserom reorganization